src/Microsoft.Diagnostics.Runtime.Tests/src/CacheOptionsTest.cs

```diff
@@ -132,8 +133,8 @@ public void FieldCachingTest()
             ClrType type = module.GetTypeByName("Foo");
             Assert.NotEqual(0ul, type.MethodTable);  // Sanity test
 
-            Assert.Same(type.Fields, type.Fields);
-            Assert.Same(type.StaticFields, type.StaticFields);
+            Assert.Equal(type.Fields, type.Fields);
```

Comment:
This isn't quite asserting the same thing.  The intent here was to assert that we only created the underlying list once and didn't recreate it every call.

Is that not possible to assert with ImmutableArray or have we actually changed the semantics, and now we recreate the immutable array every call?  If it's not possible to assert, can you assert that each entry in Fields/StaticFields are .Same?  That is close to what I was trying to do in this test.

Reply:
`ImmutableArray`s equality comparison checks the reference equality of the underlying arrays.

Reply:
Ah great!